### ISSUE TITLE
dataset_setup.py fails at Criteo1TB due to wrong folder handling

### ISSUE THREAD

**Issue by alex77g2** [mlcommons/algorithmic-efficiency#674]
dataset_setup.py fails at Criteo1TB due to wrong folder handling ( ~ handling )
For the other datasets (e.g. ogbg + wmt) the script works as expected.

## Description

```
echo $DATA_DIR/tmp # ~/data/tmp
python3 datasets/dataset_setup.py --data_dir $DATA_DIR --temp_dir $DATA_DIR/tmp --criteo1tb
...
unzip:  cannot find or open /home/user/data/tmp/criteo1tb/all_days.zip, /home/user/data/tmp/criteo1tb/all_days.zip.zip or /home/user/data/tmp/criteo1tb/all_days.zip.ZIP.
```
But I can see the file at another wrong location. "~" is handled wrongly for temp_dir , (but correct for data_dir)
**Please note 2x tilde in directory path (see next line)**
```
user@elra:~/git/algorithmic-efficiency/~/data/tmp/criteo1tb$ ls -lh
total 343G
-rw-rw-r-- 1 user user 343G Mär  3 15:51 all_days.zip
```
**unzip:  cannot find or open /home/user/data/tmp/criteo1tb/all_days.zip** should not happen, and download should be in another folder below $DATA_DIR .
The file system has > 1 TB free space.

## Steps to Reproduce
```
echo $DATA_DIR
~/data
(env_mlc) user@elra4080:~/git/algorithmic-efficiency$ python3 datasets/dataset_setup.py --data_dir $DATA_DIR --temp_dir $DATA_DIR/tmp --criteo1tb
I0303 08:06:04.740991 140054121058368 dataset_setup.py:683] Downloading data to /home/user/data...
I0303 08:06:04.741339 140054121058368 dataset_setup.py:686] Downloading criteo1tb...
I0303 08:06:09.657010 140054121058368 dataset_setup.py:287] Downloading ~342GB Criteo 1TB data .zip file:
https://download.wetransfer.com/eugv/4bbea9b4a54baddea549d71271a38e2c20230428071257/76d9a77ca7e3e01ca420bbeb8ceb04d5e5697ac7/criteo_terabyte-dataset-24-files_2023-04-28_0712.zip?cf=y&token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6ImRlZmF1bHQifQ.eyJleHAiOjE3MDk0NTAxNjksImlhdCI6MTcwOTQ0OTU2OSwiZG93bmxvYWRfaWQiOiJjZThiMjQ3OS05OGI2LTRiYTktOGI5NS1hNzQ3MGE0NDc4NTUiLCJzdG9yYWdlX3NlcnZpY2UiOiJzdG9ybSJ9.0uINdTf2UIOmHObgYVcgzUxm081NQQrGGIb-0Q7vM24
I0303 15:51:54.910727 140054121058368 dataset_setup.py:299] Running Criteo 1TB unzip command:
unzip ~/data/tmp/criteo1tb/all_days.zip -d ~/data/tmp/criteo1tb
unzip:  cannot find or open /home/user/data/tmp/criteo1tb/all_days.zip, /home/user/data/tmp/criteo1tb/all_days.zip.zip or /home/user/data/tmp/criteo1tb/all_days.zip.ZIP.
I0303 15:51:54.915950 140054121058368 dataset_setup.py:199]
```

## Source or Possible Fix

The author of the script should be able to locate the bug, as same script is fine for many other datasets.

**Comment by alex77g2** [comment]
Please also add an overwrite question for the Criteo1TB 350 GB ZIP file!
For the other datasets like FastMrt (only 90 GB) same script is asking before downloading again + overwrite.
The download takes about 5-6 hours for Criteo1TB.

**Comment by priyakasimbeg** [comment]
Thanks for raising these issues.
We can add handling for the `~` for the tmp_dir. The `os` package does not expand `~`  by default.
In the meantime, to get unblocked, please use $HOME in your arg as described in the examples in the data_setup/README.md.

**Comment by alex77g2** [comment]
I think detecting tilde and abort with msg like "dont use tilde in 2nd path" would be fine. Else it takes 6h to find out something went wrong.

**Comment by priyakasimbeg** [comment]
Requests merged into dev https://github.com/mlcommons/algorithmic-efficiency/pull/683.
Will merge into main by EOD tomorrow.

**Comment by alex77g2** [comment]
Thank you very much!
